### ISSUE TITLE
Completely remove dbus functionality?

### ISSUE THREAD

**Issue by sophron** [wifiphisher/wifiphisher#918]
It looks like dbus-python is one of the modules that it's hard to depend on.

There are Travis builds that occasionally <a href="https://travis-ci.org/wifiphisher/wifiphisher/builds/375934631?utm_source=github_status&utm_medium=notification">fail</a> and at the same time it seems to be an unnecessary overhead to build a shared library that we only use in a specific place.

I wonder if we can find another way to check if an interface is managed by the Network Manager.

**Comment by DarkReef** [comment]
dbus-python looks like deprecated and cannot be installed from pip repository, so using dbus package require to install it from distributive sources (python2-dbus on archlinux for example)
Is that serious reason for remove this package?

**Comment by sophron** [comment]
I wouldn't say the project is deprecated or abandoned. Last release was 4 days ago. But it is surely painful to depend on it. If we find another way to check if an interface is managed by the Network Manager, we can stay away from it.

**Comment by DarkReef** [comment]
@sophron Yeah, I picked the wrong word. Maybe we should try `pydbus`? On one side, it's more stable than `dbus-python` on the other we can get more dependencies...

**Comment by blackHatMonkey** [comment]
@sophron I think we should remove the package. IMHO it adds a lot of unnecessary overhead to the code. Instead I think we should document how users can do this manually. Perhaps we can compile a tutorial on the most popular network managers.

**Comment by sophron** [comment]
I pushed a commit that completely removes any dbus functionality. I used `nmcli` instead. While this is an external tool, my guess is that it is always installed when the NetworkManager daemon exists in the system (which is the case for modern Linux distros).

Wifiphisher installation is now significantly smoother.

We can also take this a step further and use `nmcli` to make our interfaces "unmanaged" by the NetworkManager when the `-iI` is provided by the end-user.
